core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -195,24 +199,47 @@ public void reloadConfig() {
           networkMonitor,
           authMonitor
       );
+      this.isConfigured = false;
+      this.stop();
     } finally {
       syncLock.unlock();
     }
   }
 
-
-  public <T> void configure(final MongoNamespace namespace,
-                            final ConflictHandler<T> conflictHandler,
+  public <T> void configure(@Nonnull final MongoNamespace namespace,
+                            @Nonnull final ConflictHandler<T> conflictHandler,
                             final ChangeEventListener<T> changeEventListener,
                             final ErrorListener errorListener,
                             final Codec<T> codec) {
+    if (conflictHandler == null) {
+      logger.warn(
+          "Invalid configuration: conflictHandler should not be null. "
+              + "The DataSynchronizer will not begin syncing until a ConflictHandler has been "
+              + "provided.");
+      return;
+    }
+
     this.errorListener = errorListener;
+
     this.syncConfig.getNamespaceConfig(namespace).configure(
         conflictHandler,
         changeEventListener,
         codec
     );
-    this.triggerListeningToNamespace(namespace);
+
+    if (!this.isConfigured) {
+      syncLock.lock();
+      try {
+        this.isConfigured = true;
+      } finally {
+        syncLock.unlock();
+      }
+      this.triggerListeningToNamespace(namespace);
+    }
+
+    if (!isRunning) {
+      this.start();
```

Comment:
I think this is causing your flakiness now. You have a BG thread running. Removing this will fix that.